lib/compute/subnetwork.js

```diff
@@ -0,0 +1,216 @@
+/*!
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module compute/vm
+ */
+
+'use strict';
+
+var format = require('string-format-obj');
+var nodeutil = require('util');
+
+/**
+ * @type {module:common/service-object}
+ * @private
+ */
+var ServiceObject = require('../common/service-object.js');
+
+/**
+ * @type {module:common/util}
+ * @private
+ */
+var util = require('../common/util.js');
+
+
+/*! Developer Documentation
+ *
+ * @param {module:region} region - Region this address belongs to.
+ * @param {string} name - Name of the subnetwork.
+ */
+/**
+ * An Instance object allows you to interact with a Google Compute Engine
+ * subnetwork.
+ *
+ * @resource [Instances and Networks]{@link https://cloud.google.com/compute/docs/instances-and-network}
+ * @resource [Instance Resource]{@link https://cloud.google.com/compute/docs/reference/v1/subnetworks}
+ *
+ * @constructor
+ * @alias module:compute/subnetwork
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var gce = gcloud.compute();
+ *
+ * var region = gce.region('region-name');
+ *
+ * var subnetwork = region.subnetwork('subnetwork1');
+ */
+function Subnetwork(region, name) {
+  this.name = name.replace(/.*\/([^/]+)$/, '$1'); // Just the instance name.
+  this.region = region;
+
+  // this.url = format('{base}/{project}/regions/{regions}/subnetwork/{name}', {
+  //   base: 'https://www.googleapis.com/compute/v1/projects',
+  //   //project: region.projectId,
+  //   zone: region.name,
+  //   name: this.name
+  // });
+
+  var methods = {
+    /**
+     * Create a subnetwork.
+     *
+     * @param {object} config - See {module:compute/region#createSubnetwork}.
```

Comment:
Oh, this will be tricky. If we remove `region.createSubnetwork`, we can't expose a create method on a Subnetwork object. I guess we should keep both `region.createSubnetwork` and `network.createSubnetwork`. Sorry for all the back and forth.